src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -607,13 +607,21 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
     const deletes = objects.savedObjects.map((object) =>
       savedObjectsClient.delete(object.type, object.id, { force: true })
     );
-    await Promise.all(deletes);
-
-    // Unset this
-    this.setState({
-      selectedSavedObjects: [],
-    });
-
+    try {
+      await Promise.all(deletes);
+      // Unset this
+      this.setState({
+        selectedSavedObjects: [],
+      });
+    } catch (error) {
```

Comment:
Shall we just return in the catch? If delete is not successful, there is no need to fetch objects and counts maybe.

Reply:
Sure, I will move the related code to the `try {}`.

Reply:
The solution is not good, in such case, the delete modal will be closed no matter if there is error or not.

Reply:
Make sense. If the user fails to delete the saved objects, the delete modal will continue to display. If the user success successfully deletes the saved objects, the delete modal will not be displayed.

Reply:
Would you mind adding a unit test to cover this line? If delete failed, the modal should still be there will notification.toast.addDanger being called and loading is false.

Reply:
Sure, I will add some unit tests.